### PR TITLE
features/infinite scroll

### PR DESCRIPTION
**Pull request by yurii-kozlov**

### CODE REVIEW

File: api/api.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import axios, {AxiosResponse} from 'axios';
+
+const BASE_URL = 'http://localhost:3002';
```
Comment (yurii-kozlov) [inline api/api.ts:3]: I tried to use the env variable (BASE_URL) but somehow it didn't work. Should I leave it like that or would ii be better to somehow replace it with the env variable?

File: components/ProductsScroll/ProductsScroll.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+  useEffect(() => {
+    document.addEventListener('scroll', scrollHandler);
+
+    return () => {
+      document.removeEventListener('scroll', scrollHandler);
+    }
+  }, [])
+
+  const scrollHandler = (e: any):void => {
```
Comment (yurii-kozlov) [inline components/ProductsScroll/ProductsScroll.tsx:26]: If I implemented the logic in the right way, could you please give me a hint how to type this event? I was puzzling over it for a long time and wasn't able to find or guess the answer

File: api/api.ts (hunk 1 above)
Comment (TechCossack) [reply api/api.ts:3]: Pls, read the NextJs documentation and use env variable https://nextjs.org/docs/basic-features/environment-variables

File: components/ProductsScroll/Computers/Computers.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  useEffect(() => {
+    getComputers(currentPage)
+      .then(response => {
+      setComputers([...computers, ...response.data.list])
+      setCurrentPage(prevState => prevState + 1)
+  })
+  }, [])
+
+  useEffect(() => {
+    if (fetching && computers.length < 60 && visibleProductSection === Goods.computers) {
```
Comment (TechCossack) [inline components/ProductsScroll/Computers/Computers.tsx:29]: this is not good, pls tell me what is going on here))

File: components/ProductsScroll/Laptops/Laptops.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  setFetching: React.Dispatch<React.SetStateAction<boolean>>
+}
+
+export const Laptops: React.FC<Props> = ({ fetching, visibleProductSection, setFetching }): ReactElement => {
+  const [laptops, setLaptops] = useState<Product[]>([]);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  useEffect(() => {
+
+    if (fetching && laptops.length < 60 && visibleProductSection === Goods.laptops) {
```
Comment (TechCossack) [inline components/ProductsScroll/Laptops/Laptops.tsx:20]: the same)

File: components/ProductsScroll/ProductsScroll.tsx [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+import { Goods } from 'enums/goods';
+import { Computers } from 'components/ProductsScroll/Computers/Computers';
+import { Monitors } from 'components/ProductsScroll/Monitors/Monitors';
+import cn from 'classnames';
+
+export const ProductsScroll: React.FC = (): ReactElement => {
+  const [fetching, setFetching] = useState(true);
+  const [visibleProductSection, setVisibleProductSection] = useState<Goods>(Goods.laptops);
+
+  const handleLaptopsVisibility = (): void => setVisibleProductSection(Goods.laptops);
```
Comment (TechCossack) [inline components/ProductsScroll/ProductsScroll.tsx:14]: and pls describe me this logic too

File: components/ProductsScroll/ProductsScroll.tsx (hunk 2 above)
Comment (TechCossack) [reply components/ProductsScroll/ProductsScroll.tsx:26]: let's discuss it in haddle

**Review by TechCossack — CHANGES_REQUESTED** [review]

**Review by TechCossack — APPROVED** [review]

**Pull request merged**